src/ds/address.h

```diff
@@ -39,4 +42,80 @@ namespace snmalloc
   {
     return reinterpret_cast<T*>(address);
   }
+
+  /**
+   * Test if a pointer is aligned to a given size, which must be a power of
+   * two.
+   */
+  template<size_t alignment>
```

Comment:
`granule` vs `alignment` as the template parameter.  I think `alignment` is more consistent with the rest of the code.